lib/request-processor.js

```diff
@@ -1,11 +1,30 @@
 const request = require('request')
-
+const os = require('os')
+const uuidv4 = require('uuid/v4')
+const logger = require('./logger')
+const assert = require('assert')
 module.exports = class RequestProcessor {
   constructor ({requestImpl}) {
     this._request = requestImpl || request
+    this._logger = logger
+  }
+
+  _log ({extensionUri, accessToken, resources, type}) {
+    this._logger.log(JSON.stringify({
+      'id': uuidv4(),
+      'type': type,
+      'user': accessToken.substr(7, 16), // get the first 16 characters of the user's access token
+      'system': `${os.hostname()}-${os.type()}`,
+      'date': new Date(),
```

Comment:
how does this get serialized by default?

Reply:
nice catch. new Date() auto serializes to the ISO string. I'm adding `.toISOstring()` in order to make this more clear.